File: graphql-test/src/main/java/com/junyss/graphqltest/api/equipment/model/entity/Equipment.java
package com.junyss.graphqltest.api.equipment.model.entity;

import javax.persistence.Entity;
import javax.persistence.Id;

import com.junyss.graphqltest.api.equipment.model.dto.request.EquipmentRequestDto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Entity
public class Equipment {

	@Id
	private String equipmentId;
	private String usedBy;
	private Integer count;
	private String newOrUsed;

	public static Equipment toEntity (EquipmentRequestDto equipmentRequestDto) {
		return new Equipment(
			equipmentRequestDto.getEquipmentId(),
			equipmentRequestDto.getUsedBy(),
			equipmentRequestDto.getCount(),
		equipmentRequestDto.getNewOrUsed());
	}

	public void updateUsedBy(String usedBy) {
		this.usedBy = usedBy;
	}

	public void updateCount(Integer count) {
		this.count = count;
	}

	public void updateNewOrUsed(String newOrUsed) {
		this.newOrUsed = newOrUsed;
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/equipment/model/entity/Equipment.kt
package com.junyharangstudy.kotlingraphqltest.api.equipment.model.entity

import jakarta.persistence.Entity
import jakarta.persistence.Id

@Entity
class Equipment(
    @Id val equipmentId: String,
    var usedBy: String,
    var count: Int,
    var newOrUsed: String
) {

    fun updateUsedBy(usedBy: String) {
        this.usedBy = usedBy
    }

    fun updateCount(count: Int) {
        this.count = count
    }

    fun updateNewOrUsed(newOrUsed: String) {
        this.newOrUsed = newOrUsed
    }
}